pkg/services/ngalert/eval/eval.go

```diff
@@ -225,7 +225,8 @@ func buildDatasourceHeaders(ctx EvaluationContext) map[string]string {
 
 	key, ok := models.RuleKeyFromContext(ctx.Ctx)
 	if ok {
-		headers["X-Rule-Uid"] = key.UID
+		headers["X-Grafana-Rule-Uid"] = key.UID
```

Comment:
We were very intentional on the `X-Rule-Uid` naming because it is consistent with dashboards, which also offer an `X-Panel-Id` and an `X-Dashboard-Uid` for queries issued by them.

I do not think we should deviate from that precedent unless we change all 3 headers at the same time. It's a very awkward developer experience to be inconsistent here, users will get confused trying to remember which headers contain `-Grafana-` and which headers don't.

Reply:
ok